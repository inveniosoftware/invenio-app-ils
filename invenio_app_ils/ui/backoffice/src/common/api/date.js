import { DateTime } from 'luxon';

export const fromBackend = stringDate => {
  return DateTime.fromISO(stringDate);
};

export const toBackend = date => {
  return date.toISO();
};

export const toHuman = date => {
  return date.toLocaleString();
};
