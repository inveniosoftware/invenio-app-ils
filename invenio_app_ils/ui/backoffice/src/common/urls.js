export const URLS = {
  ROOT: '/',
  BASENAME: '/backoffice',
  ITEM_LIST: '/items',
  ITEM_DETAILS: id => `/items/${id}`,
  LOAN_LIST: '/loans',
  LOAN_DETAILS: id => `/loans/${id}`,
};
