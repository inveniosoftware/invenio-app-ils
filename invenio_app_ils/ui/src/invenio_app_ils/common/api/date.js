import { DateTime } from 'luxon';
import _get from 'lodash/get';

export const fromISO = stringDate => {
  return DateTime.fromISO(stringDate);
};

/**
 * Converts datetime JSON string to luxon Datetime object
 * @param date
 * @returns {DateTime}
 */
export const toISO = date => {
  return date.toISO();
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDateTime = date => {
  return date.toFormat('yyyy-MM-dd HH:mm');
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDate = date => {
  return date.toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for UTC date in short format used to
 *  normalise dates for REST API
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toUTCShortDate = date => {
  if (!(typeof date === 'string' || date instanceof DateTime)) {
    throw Error('Wrong date type provided');
  }
  if (typeof date === 'string') {
    date = fromISO(date);
  }
  return date.toUTC().toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for display in tables
 */
export function dateFormatter({ col, row }) {
  const dateField = _get(row, col.field);
  return dateField ? toShortDate(fromISO(dateField)) : null;
}
