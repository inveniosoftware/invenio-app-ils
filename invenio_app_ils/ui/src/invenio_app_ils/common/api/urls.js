export const ApiURLS = {
  loans: {
    loan: '/circulation/loans/:loanPid',
    list: '/circulation/loans/',
    checkout: '/circulation/loans/checkout',
    request: '/circulation/loans/request',
  },
  items: {},
  documents: {},
  locations: {},
  users: {},
};
