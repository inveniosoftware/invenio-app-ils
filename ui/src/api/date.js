import { DateTime } from 'luxon';
import _get from 'lodash/get';
import _hasIn from 'lodash/hasIn';

/**
 * Converts datetime JSON string to luxon Datetime object
 * @param stringDate
 * @returns {DateTime}
 */
export const fromISO = stringDate => {
  return DateTime.fromISO(stringDate);
};

/**
 * Converts luxon Datetime object to ISO date string
 * @param date
 * @returns {String}
 */
export const toISO = date => {
  if (_hasIn(date, 'toISO')) {
    return date.toISO();
  }
  return date ? date : '';
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @returns {String}
 */
export const toShortDateTime = date => {
  return date instanceof DateTime ? date.toFormat('yyyy-MM-dd HH:mm') : date;
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @returns {String}
 */
export const toShortDate = date => {
  return date instanceof DateTime ? date.toFormat('yyyy-MM-dd') : date;
};

/**
 *  Serializes date for display in tables
 */
export function dateFormatter({ col, row }) {
  const dateField = _get(row, col.field);
  return dateField ? toShortDate(fromISO(dateField)) : null;
}
