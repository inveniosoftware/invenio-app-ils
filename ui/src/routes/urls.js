import { generatePath } from 'react-router-dom';

const AuthenticationBase = '/';

const AuthenticationRoutesList = {
  login: `${AuthenticationBase}login`,
  redirectUrlAfterLogin: `${AuthenticationBase}login?next=:nextUrl`,
  confirmEmail: `${AuthenticationBase}confirm-email`,
};

const AuthenticationRoutesGenerators = {
  redirectAfterLogin: nextUrl =>
    generatePath(AuthenticationRoutesList.redirectUrlAfterLogin, {
      nextUrl: nextUrl,
    }),
};

export const AuthenticationRoutes = {
  ...AuthenticationRoutesList,
  ...AuthenticationRoutesGenerators,
};

const FrontSiteBase = '/';

const FrontSiteRoutesList = {
  home: FrontSiteBase,
  patronProfile: `${FrontSiteBase}profile`,
  documentsList: `${FrontSiteBase}search`,
  documentDetails: `${FrontSiteBase}literature/:documentPid`,
  documentRequestForm: `${FrontSiteBase}request`,
  seriesDetails: `${FrontSiteBase}series/:seriesPid`,
};

const FrontSiteRoutesGenerators = {
  documentsListWithQuery: qs => `${FrontSiteRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(FrontSiteRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
  seriesDetailsFor: seriesPid =>
    generatePath(FrontSiteRoutesList.seriesDetails, {
      seriesPid: seriesPid,
    }),
};

export const FrontSiteRoutes = {
  ...FrontSiteRoutesGenerators,
  ...FrontSiteRoutesList,
};

const BackOfficeBase = '/backoffice';

const BackOfficeRoutesList = {
  home: BackOfficeBase,
  documentCreate: `${BackOfficeBase}/documents/create`,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  documentEdit: `${BackOfficeBase}/documents/:documentPid/edit`,
  documentRequestDetails: `${BackOfficeBase}/document-requests/:documentRequestPid`,
  documentRequestsList: `${BackOfficeBase}/document-requests`,
  documentsList: `${BackOfficeBase}/documents`,
  eitemCreate: `${BackOfficeBase}/eitems/create`,
  eitemDetails: `${BackOfficeBase}/eitems/:eitemPid`,
  eitemEdit: `${BackOfficeBase}/eitems/:eitemPid/edit`,
  eitemsList: `${BackOfficeBase}/eitems`,
  ilocationsCreate: `${BackOfficeBase}/internal-locations/create`,
  ilocationsEdit: `${BackOfficeBase}/internal-locations/:ilocationPid/edit`,
  itemCreate: `${BackOfficeBase}/items/create`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  itemEdit: `${BackOfficeBase}/items/:itemPid/edit`,
  itemsList: `${BackOfficeBase}/items`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  loansList: `${BackOfficeBase}/loans`,
  locationsCreate: `${BackOfficeBase}/locations/create`,
  locationsEdit: `${BackOfficeBase}/locations/:locationPid/edit`,
  locationsList: `${BackOfficeBase}/locations`,
  patronDetails: `${BackOfficeBase}/patrons/:patronPid`,
  patronsList: `${BackOfficeBase}/patrons`,
  seriesCreate: `${BackOfficeBase}/series/create`,
  seriesDetails: `${BackOfficeBase}/series/:seriesPid`,
  seriesEdit: `${BackOfficeBase}/series/:seriesPid/edit`,
  seriesList: `${BackOfficeBase}/series`,
  stats: {
    home: `${BackOfficeBase}/stats`,
  },
};

const BackOfficeRouteGenerators = {
  documentEditFor: documentPid =>
    generatePath(BackOfficeRoutesList.documentEdit, {
      documentPid: documentPid,
    }),
  documentsListWithQuery: qs => `${BackOfficeRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(BackOfficeRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
  documentRequestsListWithQuery: qs =>
    `${BackOfficeRoutesList.documentRequestsList}?q=${qs}`,
  documentRequestDetailsFor: documentRequestPid =>
    generatePath(BackOfficeRoutesList.documentRequestDetails, {
      documentRequestPid: documentRequestPid,
    }),
  eitemDetailsFor: eitemPid =>
    generatePath(BackOfficeRoutesList.eitemDetails, { eitemPid: eitemPid }),
  eitemEditFor: eitemPid =>
    generatePath(BackOfficeRoutesList.eitemEdit, { eitemPid: eitemPid }),
  eItemsListWithQuery: qs => `${BackOfficeRoutesList.eitemsList}?q=${qs}`,
  itemsListWithQuery: qs => `${BackOfficeRoutesList.itemsList}?q=${qs}`,
  itemDetailsFor: itemPid =>
    generatePath(BackOfficeRoutesList.itemDetails, { itemPid: itemPid }),
  itemEditFor: itemPid =>
    generatePath(BackOfficeRoutesList.itemEdit, { itemPid: itemPid }),
  loansListWithQuery: qs => `${BackOfficeRoutesList.loansList}?q=${qs}`,
  loanDetailsFor: loanPid =>
    generatePath(BackOfficeRoutesList.loanDetails, { loanPid: loanPid }),
  ilocationsEditFor: ilocationPid =>
    generatePath(BackOfficeRoutesList.ilocationsEdit, {
      ilocationPid: ilocationPid,
    }),
  locationsEditFor: locationPid =>
    generatePath(BackOfficeRoutesList.locationsEdit, {
      locationPid: locationPid,
    }),
  patronDetailsFor: patronPid =>
    generatePath(BackOfficeRoutesList.patronDetails, { patronPid: patronPid }),
  seriesListWithQuery: qs => `${BackOfficeRoutesList.seriesList}?q=${qs}`,
  seriesDetailsFor: seriesPid =>
    generatePath(BackOfficeRoutesList.seriesDetails, { seriesPid: seriesPid }),
  seriesEditFor: seriesPid =>
    generatePath(BackOfficeRoutesList.seriesEdit, {
      seriesPid: seriesPid,
    }),
};

export const BackOfficeRoutes = {
  ...BackOfficeRoutesList,
  ...BackOfficeRouteGenerators,
};

const AcquisitionBase = `${BackOfficeBase}/acquisition`;

const AcquisitionRoutesList = {
  orderCreate: `${AcquisitionBase}/orders/create`,
  orderDetails: `${AcquisitionBase}/orders/:orderPid`,
  orderEdit: `${AcquisitionBase}/orders/:orderPid/edit`,
  ordersList: `${AcquisitionBase}/orders`,
  vendorCreate: `${AcquisitionBase}/vendors/create`,
  vendorDetails: `${AcquisitionBase}/vendors/:vendorPid`,
  vendorEdit: `${AcquisitionBase}/vendors/:vendorPid/edit`,
  vendorsList: `${AcquisitionBase}/vendors`,
};

const AcquisitionRouteGenerators = {
  orderDetailsFor: orderPid =>
    generatePath(AcquisitionRoutesList.orderDetails, { orderPid: orderPid }),
  orderEditFor: orderPid =>
    generatePath(AcquisitionRoutesList.orderEdit, { orderPid: orderPid }),
  ordersListWithQuery: qs => `${AcquisitionRoutesList.ordersList}?q=${qs}`,
  vendorDetailsFor: vendorPid =>
    generatePath(AcquisitionRoutesList.vendorDetails, { vendorPid: vendorPid }),
  vendorEditFor: vendorPid =>
    generatePath(AcquisitionRoutesList.vendorEdit, { vendorPid: vendorPid }),
  vendorsListWithQuery: qs => `${AcquisitionRoutesList.vendorsList}?q=${qs}`,
};

export const AcquisitionRoutes = {
  ...AcquisitionRoutesList,
  ...AcquisitionRouteGenerators,
};

const ILLBase = `${BackOfficeBase}/ill`;

const ILLRoutesList = {
  libraryList: `${ILLBase}/libraries`,
  libraryDetails: `${ILLBase}/libraries/:libraryPid`,
  libraryEdit: `${ILLBase}/libraries/:libraryPid/edit`,
  libraryCreate: `${ILLBase}/libraries/create`,
  borrowingRequestList: `${ILLBase}/borrowing-requests`,
  borrowingRequestDetails: `${ILLBase}/borrowing-requests/:borrowingRequestPid`,
  borrowingRequestEdit: `${ILLBase}/borrowing-requests/:borrowingRequestPid/edit`,
  borrowingRequestCreate: `${ILLBase}/borrowing-requests/create`,
};

const ILLRoutesGenerators = {
  libraryDetailsFor: libraryPid =>
    generatePath(ILLRoutesList.libraryDetails, {
      libraryPid: libraryPid,
    }),
  libraryEditFor: libraryPid =>
    generatePath(ILLRoutesList.libraryEdit, {
      libraryPid: libraryPid,
    }),
  documentRequestsListWithQuery: qs =>
    `${BackOfficeRoutesList.documentRequestsList}?q=${qs}`,
  borrowingRequestDetailsFor: borrowingRequestPid =>
    generatePath(ILLRoutesList.borrowingRequestDetails, {
      borrowingRequestPid: borrowingRequestPid,
    }),
  borrowingRequestEditFor: borrowingRequestPid =>
    generatePath(ILLRoutesList.borrowingRequestEdit, {
      borrowingRequestPid: borrowingRequestPid,
    }),
};

export const ILLRoutes = {
  ...ILLRoutesList,
  ...ILLRoutesGenerators,
};

export const DetailsRouteByPidTypeFor = pidType => {
  switch (pidType) {
    case 'pitmid':
      return BackOfficeRouteGenerators.itemDetailsFor;
    case 'illbid':
      return ILLRoutesGenerators.borrowingRequestDetailsFor;
    default:
      throw new Error(
        `Cannot generate url to the detail page for unknown pidType: ${pidType}`
      );
  }
};
