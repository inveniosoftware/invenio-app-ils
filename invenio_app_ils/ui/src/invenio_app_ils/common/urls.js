import { generatePath } from 'react-router-dom';
import { loan as loanApi } from './api/loan';
import { item as itemApi } from './api/item';

export const FrontSiteURLS = {};

const BackOfficeBase = '/backoffice';

export const BackOfficeURLS = {
  home: BackOfficeBase,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  itemsSearch: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansSearch: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronDetails: `${BackOfficeBase}/users/:userPid`,
};

export const viewLoanDetailsUrl = loanPid => {
  return generatePath(BackOfficeURLS.loanDetails, {
    loanPid: loanPid,
  });
};

export const showAllLoansUrl = (
  documentPid,
  itemPid,
  state,
  patronPid,
  extraQuery
) => {
  const qs = loanApi.buildLoansQuery(
    documentPid,
    itemPid,
    state,
    patronPid,
    extraQuery
  );
  return `${BackOfficeURLS.loansSearch}?q=${qs}`;
};

export const viewItemDetailsUrl = itemPid => {
  return generatePath(BackOfficeURLS.itemDetails, {
    itemPid: itemPid,
  });
};

export const showAllItemsUrl = (documentPid, state, extraQuery) => {
  const qs = itemApi.buildItemsQuery(documentPid, state, extraQuery);
  return `${BackOfficeURLS.itemsSearch}?q=${qs}`;
};
