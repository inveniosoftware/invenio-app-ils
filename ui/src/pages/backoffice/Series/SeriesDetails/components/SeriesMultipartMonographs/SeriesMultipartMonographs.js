import { InfoMessage } from '@pages/backoffice';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';
import { Loader, Error, ResultsTable, SeriesAuthors } from '@components';
import { series as seriesApi } from '@api';
import { BackOfficeRoutes } from '@routes/urls';
import { SeeAllButton } from '@pages/backoffice/components/buttons';
import _get from 'lodash/get';

export default class SeriesMultipartMonographs extends Component {
  constructor(props) {
    super(props);
    this.seriesPid = props.seriesDetails.metadata.pid;
    this.seriesType = props.seriesDetails.metadata.mode_of_issuance;
    this.fetchSeriesMultipartMonographs = props.fetchSeriesMultipartMonographs;
  }

  componentDidMount() {
    this.fetchSeriesMultipartMonographs(this.seriesPid);
  }

  seeAllButton = () => {
    const path = BackOfficeRoutes.seriesListWithQuery(
      seriesApi
        .query()
        .withSerialPid(this.seriesPid)
        .qs()
    );
    return <SeeAllButton to={path} />;
  };

  viewDetails = ({ row }) => {
    return (
      <Button
        as={Link}
        to={BackOfficeRoutes.seriesDetailsFor(row.metadata.pid)}
        compact
        icon="info"
        data-test={row.metadata.pid}
      />
    );
  };

  volumeFormatter = ({ row }) => {
    // Find the series volume for the parent series
    const relation = _get(
      row,
      `metadata.relations.${this.seriesType.toLowerCase()}`,
      []
    ).find(
      relation =>
        relation.pid === this.seriesPid && relation.pid_type === 'serid'
    );
    return relation ? relation.volume : '?';
  };

  render() {
    const { multipartMonographs, isLoading, showMaxSeries, error } = this.props;
    const columns = [
      { title: '', field: '', formatter: this.viewDetails },
      { title: 'ID', field: 'metadata.pid' },
      { title: 'Volume', formatter: this.volumeFormatter },
      { title: 'Title', field: 'metadata.title' },
      {
        title: 'Authors',
        formatter: ({ row }) => <SeriesAuthors metadata={row.metadata} />,
      },
    ];
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>
          <ResultsTable
            data={multipartMonographs.hits}
            columns={columns}
            totalHitsCount={multipartMonographs.total}
            name={'multipart monographs'}
            seeAllComponent={this.seeAllButton()}
            showMaxRows={showMaxSeries}
            renderEmptyResultsElement={() => (
              <InfoMessage
                header="No multipart monographs in this series."
                content="Start with a multipart monograph to attach it to this series."
                data-test="no-results"
              />
            )}
          />
        </Error>
      </Loader>
    );
  }
}

SeriesMultipartMonographs.propTypes = {
  showMaxSeries: PropTypes.number,
};

SeriesMultipartMonographs.defaultProps = {
  showMaxSeries: 5,
};
