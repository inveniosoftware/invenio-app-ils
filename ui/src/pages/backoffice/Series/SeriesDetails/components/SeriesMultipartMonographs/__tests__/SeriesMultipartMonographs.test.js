import React from 'react';
import { shallow, mount } from 'enzyme';
import { BackOfficeRoutes } from '@routes/urls';
import { SeriesMultipartMonographsData as SeriesMultipartMonographs } from '../SeriesMultipartMonographs';
import { Settings } from 'luxon';
import { fromISO } from '@api/date';
import { Button } from 'semantic-ui-react';

jest.mock('react-router-dom');
BackOfficeRoutes.seriesDetailsFor = jest.fn(pid => `url/${pid}`);
let mockViewDetails = jest.fn();

Settings.defaultZoneName = 'utc';
const stringDate = fromISO('2018-01-01T11:05:00+01:00');

describe('SeriesMultipartMonographs tests', () => {
  let component;
  afterEach(() => {
    mockViewDetails.mockClear();
    if (component) {
      component.unmount();
    }
  });

  const series = {
    pid: 111,
    metadata: {
      pid: 111,
      mode_of_issuance: 'SERIAL',
      relations: {},
    },
  };

  it('should load the SeriesMultipartMonographs component', () => {
    const component = shallow(
      <SeriesMultipartMonographs
        seriesDetails={series}
        multipartMonographs={{ hits: [], total: 0 }}
        fetchSeriesMultipartMonographs={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch pending loans on mount', () => {
    const mockedFetchSeriesMultipartMonographs = jest.fn();
    component = mount(
      <SeriesMultipartMonographs
        seriesDetails={series}
        multipartMonographs={{ hits: [], total: 0 }}
        fetchSeriesMultipartMonographs={mockedFetchSeriesMultipartMonographs}
      />
    );
    expect(mockedFetchSeriesMultipartMonographs).toHaveBeenCalledWith(
      series.pid
    );
  });

  it('should render show a message with no series', () => {
    component = mount(
      <SeriesMultipartMonographs
        seriesDetails={series}
        multipartMonographs={{ hits: [], total: 0 }}
        fetchSeriesMultipartMonographs={() => {}}
      />
    );

    const message = component
      .find('Message')
      .filterWhere(element => element.prop('data-test') === 'no-results');
    expect(message).toHaveLength(1);
  });

  it('should render series', () => {
    const data = {
      hits: [
        {
          ID: '1',
          updated: stringDate,
          created: stringDate,
          pid: 'series1',
          metadata: {
            pid: 'series1',
            title: 'Test',
            relations: {},
          },
        },
        {
          id: '2',
          updated: stringDate,
          created: stringDate,
          pid: 'series2',
          metadata: {
            pid: 'series2',
            title: 'Test',
            relations: {},
          },
        },
      ],
      total: 2,
    };

    component = mount(
      <SeriesMultipartMonographs
        seriesDetails={series}
        multipartMonographs={data}
        fetchSeriesMultipartMonographs={() => {}}
      />
    );

    const rows = component
      .find('TableRow')
      .filterWhere(
        element =>
          element.prop('data-test') === 'series1' ||
          element.prop('data-test') === 'series2'
      );
    expect(rows).toHaveLength(2);

    const footer = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(0);
  });

  it('should render the see all button when showing only a few series', () => {
    const data = {
      hits: [
        {
          id: 1,
          updated: stringDate,
          created: stringDate,
          metadata: {
            pid: '1',
            relations: {},
            title: 'Test',
          },
        },
        {
          id: 2,
          updated: stringDate,
          created: stringDate,
          metadata: {
            pid: '2',
            relations: {},
            title: 'Test 2',
          },
        },
      ],
      total: 2,
    };

    component = mount(
      <SeriesMultipartMonographs
        seriesDetails={series}
        multipartMonographs={data}
        fetchSeriesMultipartMonographs={() => {}}
        showMaxSeries={1}
      />
    );

    const footer = component
      .find('TableFooter')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(1);
  });

  it('should go to series details when clicking on a series row', () => {
    const data = {
      hits: [
        {
          ID: '1',
          updated: stringDate,
          created: stringDate,
          pid: 'series1',
          metadata: {
            pid: 'series1',
            title: 'Title',
            series_objs: [],
          },
        },
      ],
      total: 2,
    };

    component = mount(
      <SeriesMultipartMonographs
        seriesDetails={series}
        multipartMonographs={data}
        fetchSeriesMultipartMonographs={() => {}}
        showMaxItems={1}
      />
    );

    component.instance().viewDetails = jest.fn(() => (
      <Button onClick={mockViewDetails}></Button>
    ));
    component.instance().forceUpdate();

    const firstId = data.hits[0].pid;
    component
      .find('TableCell')
      .filterWhere(element => element.prop('data-test') === `0-${firstId}`)
      .find('Button')
      .simulate('click');
    expect(mockViewDetails).toHaveBeenCalled();
  });
});
