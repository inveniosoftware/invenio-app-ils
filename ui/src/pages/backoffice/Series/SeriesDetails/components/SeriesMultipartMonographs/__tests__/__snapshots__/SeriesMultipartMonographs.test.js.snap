// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesMultipartMonographs tests should load the SeriesMultipartMonographs component 1`] = `
<Loader>
  <Error>
    <ResultsTable
      columns={
        Array [
          Object {
            "field": "",
            "formatter": [Function],
            "title": "",
          },
          Object {
            "field": "metadata.pid",
            "title": "ID",
          },
          Object {
            "formatter": [Function],
            "title": "Volume",
          },
          Object {
            "field": "metadata.title",
            "title": "Title",
          },
          Object {
            "formatter": [Function],
            "title": "Authors",
          },
        ]
      }
      currentPage={1}
      data={Array []}
      headerActionComponent={null}
      name="multipart monographs"
      paginationComponent={null}
      renderEmptyResultsElement={[Function]}
      seeAllComponent={
        <SeeAllButton
          disabled={false}
          fluid={false}
          to="/backoffice/series?q=relations.serial.pid_type:serid AND NOT (pid:111) AND relations.serial.pid:111"
        />
      }
      showAllResults={false}
      showMaxRows={5}
      singleLine={false}
      subtitle=""
      title=""
      totalHitsCount={0}
    />
  </Error>
</Loader>
`;
